src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -3807,8 +3841,8 @@ static apt_bool_t recog_stream_read(mpf_audio_stream_t *stream, mpf_frame_t *fra
 	recognizer_data_t *r = (recognizer_data_t *) schannel->data;
 	switch_size_t to_read = frame->codec_frame.size;
 	
-	// Check schannel data is not NULL
-	if (!r) {
+	// Check schannel data is not NULL or destroyed
+	if (!r || schannel->channel_destroyed) {
```

Comment:
You need to validate if schannel is NULL including the line above where it initialize `recognizer_data_t`

Reply:
Made cahnges